### ISSUE TITLE
Create container image with Dart SDK and Flutter SDK

### ISSUE THREAD

**Issue by chalin** [flutter/website#1196]
@gspencergoog - I'll need this soonish to setup tools that run over the Dart SDK, thanks. I'd be glad to try doing it myself if the new-tag-triggered auto-Docker-container-image creation works.

cc @Sfshaza

**Comment by zoechi** [comment]
I started off https://github.com/cirruslabs/docker-images-flutter.
(no idea if that is any help)

**Comment by chalin** [comment]
Thanks @zoechi. I believe this is where we get the images for website: https://github.com/flutter/website/blob/34f0d446771f6061c1d3f61472ca7a5dadd29f8a/.cirrus.yml#L2

**Comment by gspencergoog** [comment]
Yes, sorry, I'll take a look at why this isn't firing today. The way I imagine it working is that it updates the docker images each time we update the "latest" tag on the website repo.

**Comment by gspencergoog** [comment]
@zoechi , the website uses its own docker setup, built by the code in the [tools/docker](https://github.com/flutter/website/tree/master/tool/docker) directory. The image itself comes from Google Container Registry.

**Comment by gspencergoog** [comment]
BTW, I've pushed a new image with firebase-tools@4.0.3 while I look at why the tag isn't triggering, in order to unblock you.

**Comment by chalin** [comment]
Thanks.

Let me see if I have the workflow straight (/cc @filiph). Once #1197 lands, a new container image will be created from master when a new tag is created or the `latest` tag is moved. But that means that I can't test a PR that has changes affecting the build, and even once such a PR is merged into master, master will fail until the new image is created and its build is run again using the new image. Is that about right?

Might there be a more effective workflow I'm not thinking of?

**Comment by gspencergoog** [comment]
No, ideally what you'd have is a tagged docker image that `master` is using until a new `.cirrus.yml` file lands that points to a different image with a different tag, and you'd test that other image first in the PR that contains the updated `.cirrus.yml`. The workflow for you would work something like this:

  1. Edit the Dockerfile (or Gemfile, etc.) locally
  1. Build the Docker image locally to be sure it builds properly.
  1. Commit a PR with the modified Dockerfile
  1. Tag the repo
  1. Wait for the Docker image with that tag to be built and uploaded to gcr.io
  1. Create a PR with a change to the `.cirrus.yml` that updates the tagged docker image that it is looking for.
  1. Once that PR builds successfully, commit the PR.

People with @google.com access would be able to combine some of those steps: the image could be pushed manually once it builds, and the `Dockerfile` change could be committed at the same time as the `.cirrus.yml` change, so the repo tagging wouldn't be necessary (but probably should happen, just to keep things consistent).

They'd still need to tag the new Docker image with a new tag each time.  We should probably come up with a tagging scheme (I'd suggest a version number that just gets bumped).

#1197 doesn't solve the problem yet: it's just fixing some other errors in the Dockerfile, and a potential problem with the availability of the Firebase token. I'm still working on why tagging doesn't trigger a docker build as it should.

**Comment by sfshaza2** [comment]
Do we still need this, @gspencergoog?

**Comment by gspencergoog** [comment]
No, because if you do switch back to Cirrus, it will have to be the slow way that doesn't use Docker, since Patrice doesn't have access to that resource.
